diff -uNr gedit-plugins-3.30.1-ori/plugins/codecomment/codecomment.py gedit-plugins-3.30.1/plugins/codecomment/codecomment.py
--- gedit-plugins-3.30.1-ori/plugins/codecomment/codecomment.py	2019-01-09 15:51:04.546048606 +0800
+++ gedit-plugins-3.30.1/plugins/codecomment/codecomment.py	2019-01-09 15:53:00.583910403 +0800
@@ -23,7 +23,7 @@
 
 import gi
 gi.require_version('Gtk', '3.0')
-gi.require_version('GtkSource', '3.0')
+#gi.require_version('GtkSource', '3.0')
 from gi.repository import GObject, Gio, Gtk, GtkSource, Gedit
 from gpdefs import *
 
