diff -uNr gnome-shell-3.31.2-ori/js/ui/background.js gnome-shell-3.31.2/js/ui/background.js
--- gnome-shell-3.31.2-ori/js/ui/background.js	2018-12-29 13:53:02.295691272 +0800
+++ gnome-shell-3.31.2/js/ui/background.js	2018-12-29 13:55:53.201567817 +0800
@@ -717,8 +717,11 @@
                            time: FADE_ANIMATION_TIME,
                            transition: 'easeOutQuad',
                            onComplete() {
-                               oldBackgroundActor.background.run_dispose();
-                               oldBackgroundActor.destroy();
+                             oldBackgroundActor.background.toString().match(/^.object .FINALIZED/) || (
+                               oldBackgroundActor.background.run_dispose(),
+                               oldBackgroundActor.destroy()
+                                                                                       );
+
                            }
                          });
     },
