diff -uNr tracker-2.1.4-ori/meson.build tracker-2.1.4/meson.build
--- tracker-2.1.4-ori/meson.build	2018-10-22 18:17:21.144203187 +0800
+++ tracker-2.1.4/meson.build	2018-10-23 01:37:37.095640945 +0800
@@ -52,6 +52,10 @@
 add_project_arguments('-Wno-pointer-sign', language: 'c')
 add_project_arguments('-DTRACKER_COMPILATION', language: 'c')
 
+# needed for symlink() and BYTE_ORDER
+add_project_arguments('-D_BSD_SOURCE', language : 'c')
+add_project_arguments('-D_XOPEN_SOURCE=700', language : 'c')
+
 enable_fts = get_option('fts')
 
 have_network_manager = false
