diff -uNr totem-3.30.0-ori/meson.build totem-3.30.0/meson.build
--- totem-3.30.0-ori/meson.build	2018-12-15 19:00:13.848257800 +0800
+++ totem-3.30.0/meson.build	2018-12-15 19:01:33.988074117 +0800
@@ -229,7 +229,7 @@
   endif
 
   if not have_vala
-    if vala_options == 'yes' or introspection_option == 'yes'
+    if vala_option == 'yes' or introspection_option == 'yes'
       error(str)
     endif
     message(str)
